internal/cloudapi/v2/v2.go

```diff
@@ -729,33 +729,37 @@ func (h *apiHandlers) GetComposeStatus(ctx echo.Context, id string) error {
 		if err != nil {
 			return HTTPError(ErrorMalformedOSBuildJobResult)
 		}
-		// TODO: support any number of builds
-		if len(deps) != 2 {
+		if len(deps) < 2 {
 			return HTTPError(ErrorUnexpectedNumberOfImageBuilds)
 		}
 		var initResult worker.KojiInitJobResult
 		_, _, err = h.server.workers.KojiInitJobStatus(deps[0], &initResult)
 		if err != nil {
 			return HTTPError(ErrorMalformedOSBuildJobResult)
 		}
-		var buildResult worker.OSBuildKojiJobResult
-		buildJobStatus, _, err := h.server.workers.OSBuildKojiJobStatus(deps[1], &buildResult)
-		if err != nil {
-			return HTTPError(ErrorMalformedOSBuildJobResult)
+		var buildJobResults []worker.OSBuildKojiJobResult
+		var buildJobStatuses []ImageStatus
+		for i := 1; i < len(deps); i++ {
+			var buildJobResult worker.OSBuildKojiJobResult
+			buildJobStatus, _, err := h.server.workers.OSBuildKojiJobStatus(deps[i], &buildJobResult)
+			if err != nil {
+				return HTTPError(ErrorMalformedOSBuildJobResult)
+			}
+			buildJobResults = append(buildJobResults, buildJobResult)
+			buildJobStatuses = append(buildJobStatuses, ImageStatus{
+				Status: imageStatusFromKojiJobStatus(buildJobStatus, &initResult, &buildJobResult),
+			})
 		}
 		response := ComposeStatus{
 			ObjectReference: ObjectReference{
 				Href: fmt.Sprintf("/api/image-builder-composer/v2/composes/%v", jobId),
 				Id:   jobId.String(),
 				Kind: "ComposeStatus",
 			},
-			Status: composeStatusFromKojiJobStatus(finalizeStatus, &initResult, []worker.OSBuildKojiJobResult{buildResult}, &result),
-			ImageStatuses: &[]ImageStatus{
-				{
-					Status: imageStatusFromKojiJobStatus(buildJobStatus, &initResult, &buildResult),
-				},
-			},
-			KojiStatus: &KojiStatus{},
+			Status:        composeStatusFromKojiJobStatus(finalizeStatus, &initResult, buildJobResults, &result),
+			ImageStatus:   buildJobStatuses[0], // backwards compatibility
```

Comment:
That's the opposite of @Gundersanne comment on the first version ;) It would technically speaking be a breaking change, so I changed it to keep `ImageStatus` a mandatory field.

Reply:
I think non-breaking changes should become a habit :)

Reply:
haha sorry then :D